centreon/www/include/configuration/configObject/host/DB-Func.php

```diff
@@ -1626,7 +1626,10 @@ function updateHost($host_id = null, $from_MC = false, $cfg = null)
                 $httpClient
             );
 
-            if (! empty($passwordTypeData)) {
+            if (! array_key_exists('host_snmp_community_is_password', $ret) && ! empty($hostSecrets)) {
+                // If no more fields are password types, we delete the host from the vault has it will not be readen.
```

Comment:
Past tense of "read" is "read"
```suggestion
                // If no more fields are password types, we delete the host from the vault has it will not be read.
```